op-challenger2/game/fault/types/position.go

```diff
@@ -154,3 +154,17 @@ func bigMSB(x *big.Int) Depth {
 	}
 	return Depth(x.BitLen() - 1)
 }
+
+func (p Position) MoveN(depth uint64, branch uint64) Position {
```

Comment:
I think the key component of MSFDG is to calculate depth and index correctly, here deserves some comment about `depth`/`branch` and why `indexAtDepth` is calculated this way for clarify.